make/common/NativeCompilation.gmk

```diff
@@ -216,6 +216,19 @@ DEPENDENCY_TARGET_SED_PATTERN := \
 # dependency files to contain paths compatible with make.
 #
 ifeq ($(ALLOW_ABSOLUTE_PATHS_IN_OUTPUT)-$(FILE_MACRO_CFLAGS), false-)
+  NEED_FIX_DEPS_FILE = 1
+endif
+
+# CCACHE_BASEDIR needs fix-deps-file as makefiles use absolute filenames for
+# object files while CCACHE_BASEDIR will make ccache relativize all paths for
+# its compiler. The compiler then produces relative dependency files.
+# make does not know a relative and absolute filename is the same so it will
+# ignore such dependencies.
+ifneq ($(CCACHE), )
+  NEED_FIX_DEPS_FILE = 1
```

Comment:
If the ccache case needs both of these macros, then the variable name should reflect that better. Maybe `REWRITE_PATHS_RELATIVE`?